ssz/tree_hash/tree_hash.py

```diff
@@ -30,19 +32,38 @@
 
 
 def hash_tree_root(input_object: Any, sedes: Any=None) -> Hash32:
+    result = _hash_tree_root(input_object, sedes)
+    if len(result) < 32:
+        return Hash32(result.ljust(32, b'\x00'))
+    return Hash32(result)
+
+
+def _hash_tree_root(input_object: Any, sedes: Any=None) -> bytes:
```

Comment:
To me it seems like it's not worth separating this into two functions as `ljust` is only executed in one branch.

Reply:
Since for the integer type, the intermediate result is not ljusted while the end result is. Separating hash_tree_root into two functions is one way to tell when to use the end result.